packages/common/src/createContract.ts

```diff
@@ -1,173 +1,4 @@
-            options: WriteContractParameters
-          ): Promise<WriteContractParameters<TAbi, typeof functionName, TChain, TAccount>> {
-            if (options.gas) {
-              debug("gas provided, skipping simulate", functionName, options);
-              return options as unknown as WriteContractParameters<TAbi, typeof functionName, TChain, TAccount>;
-            }
-
-            debug("simulating write", functionName, options);
-            const { request } = await publicClient.simulateContract({
-              ...options,
-              account: options.account ?? walletClient.account,
-            } as unknown as SimulateContractParameters<TAbi, typeof functionName, TChain>);
-
-            return request as unknown as WriteContractParameters<TAbi, typeof functionName, TChain, TAccount>;
-          }
-
-          async function write(options: WriteContractParameters): Promise<Hex> {
-            const preparedWrite = await prepareWrite(options);
-
-            return await pRetry(
-              async () => {
-                if (!nonceManager.hasNonce()) {
-                  await nonceManager.resetNonce();
-                }
-
-                const nonce = nonceManager.nextNonce();
-                debug("calling write function with nonce", nonce, preparedWrite);
-                return await walletClient.writeContract({
-                  nonce,
-                  ...preparedWrite,
-                });
-              },
-              {
-                retries: 3,
-                onFailedAttempt: async (error) => {
-                  // On nonce errors, reset the nonce and retry
-                  if (nonceManager.shouldResetNonce(error)) {
-                    debug("got nonce error, retrying", error);
-                    await nonceManager.resetNonce();
-                    return;
-                  }
-                  // TODO: prepareWrite again if there are gas errors?
-                  throw error;
-                },
-              }
-            );
-          }
-
-          return (...parameters) => {
-            const id = `${walletClient.chain.id}:${walletClient.account.address}:${nextWriteId++}`;
-            const { args, options } = <
-              {
-                args: unknown[];
-                options: UnionOmit<WriteContractParameters, "address" | "abi" | "functionName" | "args">;
-              }
-            >getFunctionParameters(parameters as any);
-
-            const request = {
-              address,
-              abi,
-              functionName,
-              args,
-              ...options,
-            };
-
-            const result = write(request);
-
-            onWrite?.({ id, request, result });
-
-            return result;
-          };
-        },
-      }
-    );
-  }
-
-  return contract as unknown as GetContractReturnType<TAbi, TPublicClient, TWalletClient, TAddress>;
-}
+/** @deprecated use `getContract` instead */
+export const createContract = getContract;
```

Comment:
Slightly different approach to what we've been doing in the past to avoid a breaking change. We can just remove this in a future version as a breaking change.